### ISSUE TITLE
Cluster Setup Information

### ISSUE THREAD

**Issue by deaerator** [FezVrasta/ark-server-tools#927]
Simply put how to set it using docker. I have setup 4 docker Ark servers and would like to create a cluster.  No clear information on how to do this with a docker.

**Comment by josephcy95** [comment]
Go to **/etc/arkmanager/arkmanager.cfg** (or for instances basis go into the .cfg in instances)

Then add this few line to it
```
arkflag_NoTransferFromFiltering=""  
arkopt_clusterid=CLUSTERID  #An ID that use to identify clusters, make sure to use a unique id
```

**Comment by deaerator** [comment]
That doesn't work.  I understand now that all of these servers need to be ran under one docker in order for it to work.  Apparently they need to have the same Server PID.  And that is something way beyond my understanding of Linux.

**Comment by deaerator** [comment]
Can someone tell me if this correct syntax to use. 

arkflag_NoTransferFromFiltering="False"
arkopt_clusterid=******
arkflag_PreventDownloadSurvivors=False
arkflag_PreventDownloadItems=False
arkflag_PreventDownloadDinos=False
arkflag_PreventUploadSurvivors=False
arkflag_PreventUploadItems=False
arklfag_PreventUploadDinos=False
arkopt_ClusterDirOveride=/Saves

**Comment by klightspeed** [comment]
With the arkflag_* settings, only the presence of the setting is used, not its value.

**Comment by deaerator** [comment]
I can now have the ability to see a potential transfer but I do not see the second Ark. 
I have now put the following in the start command but unsure if this is correct.

ark_NoTransferFromFiltering="False"
ark_ClusterDirOveride=/Saves
ark_clusterid=Mr.David.S.Pumpkins
.
I put the Following in the arkmanager.cfg
```
ark_PreventDownloadSurvivors=False
ark_PreventDownloadItems=False
ark_PreventDownloadDinos=False
ark_PreventUploadSurvivors=False
ark_PreventUploadItems=False
arkl_PreventUploadDinos=False
ark_ClusterDirOveride=/Saves

```

>

**Comment by klightspeed** [comment]
`ark_NoTransferFromFiltering="False"` adds the option `?NoTransferFromFiltering=False` to the server options string, and would probably enable TransferFromFiltering.

In order to add `-NoTransferFromFiltering` to the ARK server commandline, you need to use `arkflag_NoTransferFromFiltering=`*anything* where `anything` could be `""` or `true` or anything.

In order to add `-ClusterDirOverride=/Saves` to the ARK server commandline, you need to use `arkopt_ClusterDirOverride=/Saves`, and make sure that `/Saves` directory exists and is writable by the user under which you are running the servers.

In order to add `-clusterid=Mr.David.S.Pumpkins` to the ARK server commandline, you need to use `arkopt_clusterid=Mr.David.S.Pumpkins`.

Note that the servers need to run under the same UID - i.e. under the same user.  Inter-process communication and inter-server transfers won't work if the servers are running under different users.

**Comment by deaerator** [comment]
I guess that's my problem. This was designed to work in a full linux OS, and I'm using a docker setup.  I can get multiple servers up and running with common mounts for saves and cluster IDs but just can't get the same UID on each docker.  I will try this in a vm if i can't figure this out.

**Comment by klightspeed** [comment]
A docker instance is essentially a VM with a config to bring it to a known state.  I would think it'd be possible to modify the startup script to put instance files in place.

I don't know if the ARK server uses inter-process communication (in which case the instances need to be running in the same VM) or normal files to communicate between them.

**Comment by deaerator** [comment]
If I knew how to setup multiple servers in a single docker using one config file would be the perfect setup.  But that is something beyond my comprehension on how to do it.  I wish it was easy to do.  Using this guys docker file https://hub.docker.com/r/bigdaddymccarron/ark/.

**Comment by Al3601** [comment]
i have 2 ark servers runinng with docker, to setup the cluster , you just need to create à tunnel and tell your 2 docker image to join the same network gateaway (usually 172.18.0.1...), then you are done

**Comment by deaerator** [comment]
Being extremely new to linux, how do you do that?

**Comment by Al3601** [comment]
for example, you need to known what the name of the network of your first ark container, then execute this command : docker run -itd --network=YOURFIRSTARKNETWORKNAME ArkServer2

- YOURFIRSTARKNETWORKNAME = the name of the network of your 1st server container

- ArkServer2 = The name of your second ark container (not the network name)

**Comment by deaerator** [comment]
Network ID or ID; I see a large list of numbers and letters in small caps. When I try to run that command error. 
docker: Error response from daemon: pull access denied for 638f60282845, repository does not exist or may require 'docker login'.
See 'docker run --help'.

**Comment by deaerator** [comment]
Does it matter if the container is running on host or bridge.  I get a lot funky errors when I do host.

**Comment by deaerator** [comment]
How do I find this name YOURFIRSTARKNETWORKNAME, are you talking about the network named Bridge or host or is there another YOURFIRSTARKNETWORKNAME ?

**Comment by Al3601** [comment]
@deaerator NO, i don't have your docker image, so i cannot tell you the name, bridge or host or none.

- docker images    (this cmd will tell you the names of all the images you have installed)

**Comment by deaerator** [comment]
Does it matter if you place some settings in the arkmanager.cfg or the startup cmd.

**Comment by deaerator** [comment]
Close This Issue.  Apparently my Server Config with Docker and the arkmanager.cfg had nothing to do with my problem of not building a Cluster.  My problem is my ISP and router not supporting **NAT Loopback.**  My isp doesn't support this feature on their routers.  Thank you everyone for their time and help on the matter.

**Comment by JulianGogK24** [comment]
you can as well specify it in a docker-compose.yml if both docker containers are running in the same specified network e.g. ark_net these containers should be able to "find" each other by name as well ;) 

e.g.:

```
version: '3'

networks:
  ark_net:

services:
  ark_instance_1:
    container_name: arkserver_instance_1
    image: ....
    ....
    ....
    ports:
      - 7778:7778/udp
      - 7778:7778
      - 27015:27015/udp
      - 27015:27015
      - 32330:32330
    networks:
      - ark_net

 ark_instance_2:
    container_name: arkserver_instance_2
    image: ....
    ....
    ....
    ports:
      - 7778:7778/udp
      - 7778:7778
      - 27015:27015/udp
      - 27015:27015
      - 32330:32330
    networks:
      - ark_net

```
